packages/account/src/Components/additional-kyc-info-modal/additional-kyc-info-modal.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import { observer, useStore } from '@deriv/stores';
+import {
+    Modal,
+    DesktopWrapper,
+    MobileWrapper,
+    UILoader,
+    PageOverlay,
+    Div100vhContainer,
+    Text,
+    InlineMessage,
+} from '@deriv/components';
+import { Localize } from '@deriv/translations';
+import AdditionalKycInfoForm from './additional-kyc-info-form';
+
+type TAdditionalKycInfoFormWithHintBox = {
+    error?: unknown;
+    setError?: React.Dispatch<React.SetStateAction<unknown>>;
+};
+
+const AdditionalKycInfoFormWithHintBox = ({ error, setError }: TAdditionalKycInfoFormWithHintBox) => {
+    return (
+        <React.Fragment>
+            {!!error && (
+                <InlineMessage
+                    size='sm'
+                    message={
+                        error &&
+                        typeof error === 'object' &&
```

Comment:
**NIT:** This can be converted to a util function as these kind of checks are being used in a lot of forms to explicitly tell TS that a prop exists in an object

Reply:
I would say @farzin-deriv is a better person who can answer this question as this is kinda global for the app

Reply:
@shahzaib-deriv @likhith-deriv You guys can add utility type predicates functions to `@deriv/utils` package and use them around, Just make sure you add generic things to `@deriv/utils` and not things specific to a use case.

Example of a user-defined type predicates functions that checks if a error object is a server error or not and then infer the correct type:
```ts
type TServerError = {
    code: string;
    message: string;
    details?: { [key: string]: string };
    fields?: string[];
};

const isServerError = (error: unknown): error is TServerError => {
    return (
        error instanceof Object &&
        'code' in error &&
        'message' in error &&
        typeof error.code === 'string' &&
        typeof error.message === 'string'
    );
};
```

Usage:
```ts
import { isServerError } from '@deriv/utils`;

const { error } = useFetch('foo');

if (isServerError(error)) {
  // Can access the `message` property with the type of `string` inside the if body.
  console.log(error.message);
}
```

Reply:
Thanks @farzin-deriv for your input
Is it recommended to add this util function now?

Reply:
It's @likhith-deriv call 👀